musl versions before 1.2.2 don't have all the required fanotify features,
so a vendored header is used, but it redefines some types and the build fails
--- a/libinotifytools/src/inotifytools/fanotify-dfid-name.h
+++ b/libinotifytools/src/inotifytools/fanotify-dfid-name.h
@@ -100,42 +100,10 @@
 
 #define FANOTIFY_METADATA_VERSION 3
 
-struct fanotify_event_metadata {
-	__u32 event_len;
-	__u8 vers;
-	__u8 reserved;
-	__u16 metadata_len;
-	__aligned_u64 mask;
-	__s32 fd;
-	__s32 pid;
-};
-
 #define FAN_EVENT_INFO_TYPE_FID 1
 #define FAN_EVENT_INFO_TYPE_DFID_NAME 2
 #define FAN_EVENT_INFO_TYPE_DFID 3
 
-/* Variable length info record following event metadata */
-struct fanotify_event_info_header {
-	__u8 info_type;
-	__u8 pad;
-	__u16 len;
-};
-
-/* Unique file identifier info record */
-struct fanotify_event_info_fid {
-	struct fanotify_event_info_header hdr;
-	__kernel_fsid_t fsid;
-	/*
-	 * Following is an opaque struct file_handle that can be passed as
-	 * an argument to open_by_handle_at(2).
-	 */
-	unsigned char handle[];
-};
-
-struct fanotify_response {
-	__s32 fd;
-	__u32 response;
-};
 
 /* Legit userspace responses to a _PERM event */
 #define FAN_ALLOW 0x01
